File: intro/IncreasingArray.java
import java.util.*;

public class IncreasingArray {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int temp = s.nextInt();
        long moves = 0;

        for (int i = 1; i < n; i++) {
            int cur = s.nextInt();
            moves += Math.max(0, temp - cur);
            temp = cur + Math.max(0, temp - cur);
        }

        System.out.println(moves);
    }
}

File: intro/IncreasingArray.cpp
#include <iostream>
using namespace std;
typedef long long ll;

int main() {
    ll n, temp;
    cin >> n;
    cin >> temp;
    ll res = 0;

    for (int i = 1; i < n; i++) {
        ll cur;
        cin >> cur;
        res += max((ll) 0, temp - cur);
        temp = cur + max((ll) 0, temp - cur);
    }

    cout << res << endl;
}